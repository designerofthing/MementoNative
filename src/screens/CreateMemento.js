import { API, graphqlOperation } from "aws-amplify";
import React, { useState } from "react";
import {
  View,
  Text,
  TouchableOpacity,
  TextInput,
  StyleSheet,
} from "react-native";
import { createMementoModel } from "../../graphql/mutations";
import AppHeader from "../components/AppHeader";
import ImagePickerComponent from "../components/ImagePickerComponent";

const CreateMemento = ({ navigation }) => {
  const [mementoTitle, setMementoTitle] = useState("");
  const [mementoDescription, setMementoDescription] = useState("");
  const [fileURL, setFileURL] = useState('');

  const handleTitle = (text) => {
    setMementoTitle(text);
  };
  const handleDescription = (text) => {
    setMementoDescription(text);
  };

  const handleUploadImage = (file) => {
    console.log(file);
    setFileURL(file);
    
    
    // const formData = new FormData();
    // formData.append('file', file);
    // const newFiles = [...this.state.fileURL]
    //     this.fileArray.push(URL.createObjectURL(file))
    //     newFiles.push(file)
    // this.setState({fileURL: newFiles}, () => {

    // });

};

  const handleSubmit = async (mementoTitle, mementoDescription, fileURL) => {
    let Title = mementoTitle;
    let Description = mementoDescription;
    let ProfileImage = fileURL;
    console.log(ProfileImage);
    const input = { Title, Description, ProfileImage };
    if (Title.length !== 0 && Description.length !== 0 && ProfileImage !== undefined) {
      try {
        await API.graphql(
          graphqlOperation(createMementoModel, { input: input })
        );
        alert(Title + "'s Memento created Successfully! ");
        navigation.navigate("Home");
      } catch (err) {
        console.log("error creating memento:" + err);
      }
    } else {
      alert("Please complete all the fields");
    }
  };

  return (
    <View>
      <AppHeader />
      <TextInput
        style={styles.input}
        underlineColorAndroid="transparent"
        placeholder=" Loved one's name for memento"
        placeholderTextColor="#50055E"
        autoCapitalize="none"
        allowFontScaling
        onChangeText={handleTitle}
      />

      <TextInput
        style={styles.input}
        required
        underlineColorAndroid="transparent"
        placeholder="A short description of them"
        placeholderTextColor="#50055E"
        autoCapitalize="none"
        allowFontScaling
        onChangeText={handleDescription}
      />
      <ImagePickerComponent sendFile={handleUploadImage}/>

      <TouchableOpacity
        style={styles.submitButton}
        onPress={() =>
          handleSubmit(
            mementoTitle,
            mementoDescription,
            fileURL
          )
        }
      >
        <Text style={styles.submitButtonText}> Submit </Text>
      </TouchableOpacity>
    </View>
  );
};

export default CreateMemento;

const styles = StyleSheet.create({
  container: {
    paddingTop: 23,
  },
  input: {
    textAlign: "center",
    margin: 15,
    height: 40,
    borderColor: "#50055E",
    borderWidth: 1,
    borderRadius: 5,
  },
  submitButton: {
    textAlign: "center",
    backgroundColor: "#50055E",
    padding: 10,
    margin: 15,
    height: 40,
    borderRadius: 5,
  },
  submitButtonText: {
    color: "white",
    fontSize: 16,
  },
});
