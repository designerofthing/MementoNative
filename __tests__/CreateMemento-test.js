import React from 'react';
import {FlatList, Text, TextInput} from 'react-native';
import CreateMemento from '../src/screens/CreateMemento';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Create Memento', () => {
  jest.setTimeout(15000);
  const handleSubmit = jest.fn();
  const handleTitle = jest.fn();
  const handleDescription = jest.fn();
  const handleUploadImage = jest.fn();
  const wrapper = <CreateMemento mementoDescription='This is the test description' />
  const mementoTitle = 'Test Title';
  const fileURL = "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEAYABgAADâ€¦ouhb3R2UYtyi3HssXceFuqMTGI8sPk8hjO8LeZfOUqj//2Q=="
  
  
  it('renders correctly', async () => {
    const tree = renderer.create(<CreateMemento />);
    expect(tree).toMatchSnapshot('CreateMemento-test.js.snap');
  });
  
  
  it('renders the TextInput component', () => {
    const tree = renderer
      .create(<TextInput autoCorrect={false} value="apple banana kiwi" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});