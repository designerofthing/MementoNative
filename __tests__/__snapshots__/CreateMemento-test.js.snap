// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Memento renders correctly: CreateMemento-test.js.snap 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 90,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#50055E",
          "fontSize": 60,
        }
      }
    >
      memento
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder=" Loved one's name for memento"
    placeholderTextColor="#50055E"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#50055E",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="A short description of them"
    placeholderTextColor="#50055E"
    rejectResponderTermination={true}
    required={true}
    style={
      Object {
        "borderColor": "#50055E",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#50055E",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 40,
            "justifyContent": "center",
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#50055E",
              "fontSize": 16,
            }
          }
        >
          Select a profile image
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#50055E",
        "borderRadius": 5,
        "height": 40,
        "margin": 15,
        "opacity": 1,
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
       Submit 
    </Text>
  </View>
</View>
`;

exports[`Create Memento renders the TextInput component 1`] = `
<TextInput
  allowFontScaling={true}
  autoCorrect={false}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
  value="apple banana kiwi"
/>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 90,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#50055E",
          "fontSize": 60,
        }
      }
    >
      memento
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder=" Loved one's name for memento"
    placeholderTextColor="#50055E"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#50055E",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="A short description of them"
    placeholderTextColor="#50055E"
    rejectResponderTermination={true}
    required={true}
    style={
      Object {
        "borderColor": "#50055E",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#50055E",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 40,
            "justifyContent": "center",
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#50055E",
              "fontSize": 16,
            }
          }
        >
          Select a profile image
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#50055E",
        "borderRadius": 5,
        "height": 40,
        "margin": 15,
        "opacity": 1,
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
       Submit 
    </Text>
  </View>
</View>
`;
